packages/spindle-ui/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback } from 'react';
+import MenuHorizotal from '../Icon/MenuHorizotal';
+
+import PaginationNavigation from './PaginationNavigation';
+import { useShowItems } from './hooks/useShowItems';
+import { createUrl } from './util/url';
+
+interface Props extends React.HTMLAttributes<HTMLElement> {
+  url: string; // TODO: example url={`/article/page-PAGE_NUMBER.html`}
+  current: number;
+  total: number;
+  showCount?: boolean;
+  showPrevNext?: boolean;
+  showFirstLast?: boolean;
+  onPagination: (event: any, pageNumber: number) => void;
+}
+
+const BLOCK_NAME = 'spui-Pagination';
+
+export const Pagination = (props: Props) => {
+  const {
+    url,
+    current,
+    total,
+    showCount = false,
+    showPrevNext = true,
+    showFirstLast = false,
+    onPagination,
+    className,
+  } = props;
+
+  const pageItems = 5;
+  const { showItems, showPrevHorizotal, showNextHorizotal } = useShowItems({
+    current,
+    total,
+    pageItems,
+  });
+
+  const handleOnClick = useCallback((event, pageNumber: number) => {
+    onPagination && onPagination(event, pageNumber);
```

Comment:
```suggestion
    onPagination?.(event, pageNumber);
```
IMO. こちらのように短く書くこともできそうです！